### DISCUSSION TITLE
Retrieve a features metadata

### DISCUSSION THREAD

**Discussion by sievins** [featurehub-io/featurehub-javascript-sdk#195] [Q&A]
I can't seem to retrieve a features meta data. Is this supported via the SDK?

Thanks 🙇 .

**Comment by rvowles** [comment]
Hi there! No, the metadata is only supported by the ADK which is here: https://github.com/featurehub-io/featurehub-javascript-adk and has some documentation here (in Java, hut the js is pretty much the same) https://docs.featurehub.io/featurehub/latest/admin-development-kit.html

Did you have a use case you would like  to share?

**Reply by RenatoMPestana** [reply]
Hi @rvowles , we mainly tried to capture the feature name, as we noticed that wasnt returned by the API, we tried getting the metadata, but thats not available either. We mainly can work with the feature name alongside the key to be returned by the API, so we can filter some out based on specific conditions on the name while leaving the key intact, is that something that can be achieved somehow with the sdk? Or do we need the ADK for that aswell?

**Reply by sievins** [reply]
Thanks @RenatoMPestana and @rvowles.

Yep, so we are basically looking to group features on front end, either by name or some meta data property.

While this might be achievable via the ADK we don't want to use it on the front end and give it access to an admin service account.

**Comment by IrinaSouth** [comment]
Hi @RenatoMPestana , @sievins  - not sure if relevant to your use case, there is also an option for Feature Groups (when you group by a common strategy). It's a new feature we added in 1.7.0 https://docs.featurehub.io/featurehub/latest/feature-groups.html

**Comment by rvowles** [comment]
We seem to be out of timezone with each other otherwise I would suggest a catch up by video conference, but the current SDK API is documented here: https://github.com/featurehub-io/featurehub/blob/main/infra/api-bucket/files/edge/1.1.6.yaml - and unfortunately neither of those things is available to you.

It would seem that having another JSON feature that is auto-updated with the info you require might do what you want? In which case the webhook API might work for you?

**Reply by sievins** [reply]
Thanks @rvowles

Ha, we seem to be in literally opposite time zones. If you're happy to start a bit early we could do an 8am/8pm video conference? Is your GitHub email the best way to DM you?

@IrinaSouth's suggestion might work assuming that the feature groups are available from the SDK (on the [strategies property?](https://github.com/featurehub-io/featurehub/blob/main/infra/api-bucket/files/edge/1.1.7.yaml#L561)).

**Reply by rvowles** [reply]
Sorry - have been away, yes - those times are perfect. Pinging me on our Slack is also an option as we are normally available all the way through to 10pm (our time).

**Reply by sievins** [reply]
Thanks @rvowles. We've invited you to a Slack channel in our org via Slack Connect, so we can chat further. You should have an email.

Thanks for the support.
